internal/core/server/kernel.go

```diff
@@ -116,10 +117,24 @@ func (kern *Kernel) Ping(context.Context, *api.PingInput) (*api.PingOutput, erro
 	return &api.PingOutput{}, nil
 }
 
+func (kern *Kernel) Exit(context.Context, *api.ExitInput) (*api.ExitOutput, error) {
+	// Return immediately, then asynchronously try to shutdown.
+	defer func() {
+		go func() {
```

Comment:
What is the purpose of the `defer func() { go func() {` here? Seems like simply `go func() {` would do essentially the same thing. If the `go` was causing it to exit _before_ returning the response to a client, that's a race condition. Seems like the `defer` here is akin to `sleep`.

I suspect that the right thing to do here is to actually make `Exit` a non-josh method and instead make a custom HTTP route/handler for it. This way, the server can be sure to return a 200 OK to the client _before_ starting to exit.

Reply:
Yes, this does essentially function the same as a sleep. I'm happy to make a separate handler for it, but right now the only means we have of shutting down cleanly is externally through a process signal rather than some handle that is passed down into the `Kernel` struct. I'll remove the defer for now since it probably has no effect in most cases, and rely on the call to `svr.Shutdown(ctx)` flush this connection before shutting down.